dotCMS/src/main/java/com/dotmarketing/portlets/htmlpageasset/business/render/HTMLPageAssetRenderedAPI.java

```diff
@@ -37,52 +34,35 @@
      * @throws DotDataException     An error occurred when accessing the data source.
      * @throws IOException
      */
-    public PageView getPageMetadata(final HttpServletRequest request, final HttpServletResponse response,
-                                    final User user, final String uri, final PageMode mode)
-            throws DotSecurityException, DotDataException;
+     PageView getPageMetadata(
+            final PageContext context,
+            final HttpServletRequest request,
+            final HttpServletResponse response)
+                throws DotSecurityException, DotDataException;
 
     /***
      * Returns the rendered version of an HTML Page, i.e., the HTML code that will be rendered in
      * the browser.
      *
-     * @param request The {@link HttpServletRequest} object.
-     * @param response The {@link HttpServletResponse} object.
-     * @param user The {@link User} performing this action.
-     * @param pageUri The path to the HTML Page whose information will be retrieved.
-     * @param pageMode
+     * @param context The {@link PageContext} object.
      * @return The {@link HTMLPageAssetRendered} object containing the metadata of the different objects that
      * make up an HTML Page.
      * @throws DotSecurityException The user does not have the specified permissions to perform
      *                              this action.
      * @throws DotDataException     An error occurred when accessing the data source.
      * @throws IOException
      */
-    public PageView getPageRendered(final HttpServletRequest request, final HttpServletResponse response,
-                                    final User user, final String pageUri, final PageMode pageMode)
-            throws DotDataException, DotSecurityException;
```

Comment:
let's deprecate and add comment to use new one instead - and don't remove the old one yet